velox/exec/tests/NestedLoopJoinTest.cpp

```diff
@@ -255,6 +255,64 @@ TEST_F(NestedLoopJoinTest, basicCrossJoin) {
       "SELECT * FROM t, (SELECT * FROM UNNEST (ARRAY[10, 17, 10, 17, 10, 17, 10, 17])) u");
 }
 
+TEST_F(NestedLoopJoinTest, crossJoinWithoutFilterAndCriteria) {
+  auto probeVectors = {
+      makeRowVector({sequence<int32_t>(10)}),
+      makeRowVector({sequence<int32_t>(100, 10)}),
+  };
+
+  auto buildVectors = {
+      makeRowVector({sequence<int32_t>(1, 111)}),
+  };
+
+  createDuckDbTable("t", {probeVectors});
+  createDuckDbTable("u", {buildVectors});
+
+  // Left join.
+  auto planNodeIdGenerator = std::make_shared<core::PlanNodeIdGenerator>();
```

Comment:
Since the same plan/SQL test is repeated 3 times, you can abstract it in a lambda that takes joinType and joinConditionSQL string as parameters and invoke that 3 times with the different join types.